logicrunner/goplugin/preprocessor/main.go

```diff
@@ -1,67 +1,109 @@
 package main
 
 import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/ast"
 	"go/parser"
 	"go/token"
-	"log"
-
+	"io"
+	"io/ioutil"
 	"os"
-
-	"go/ast"
-
 	"strings"
 
-	"bytes"
-	"io"
-
-	flag "github.com/spf13/pflag"
+	"github.com/pkg/errors"
 )
 
-func init() {
-	flag.Parse()
+type ContractInterface struct {
+	Types    map[string]string
+	Methods  map[string][]*ast.FuncDecl
+	Contract string
 }
 
+var mode string
+var outfile string
+
 func main() {
-	log.Println(os.Getwd())
+	flag.StringVar(&mode, "mode", "wrapper", "Generation mode: <wrapper|helper>")
+	flag.StringVar(&outfile, "o", "-", "output file")
+	flag.Parse()
+
+	var output io.WriteCloser
+	if outfile == "-" {
+		output = os.Stdout
+	} else {
+		var err error
+		output, err = os.OpenFile(outfile, os.O_WRONLY, 0644)
+		if err != nil {
+			panic(err)
+		}
+		defer output.Close()
+	}
+
 	for _, fn := range flag.Args() {
-		w := generateForFile(fn)
-		_, _ = io.Copy(os.Stdout, w)
+		w, err := generateForFile(fn)
+		if err != nil {
+			panic(err)
+		}
+		if _, err = io.Copy(output, w); err != nil {
+			panic("Something weird with output: ", err)
```

Comment:
invalid operation: too many arguments for panic("Something weird with output: ", err) (expected 1, found 2)